lisa/lisa-analyses/src/main/java/it/unive/lisa/analysis/string/Brick.java

```diff
@@ -0,0 +1,232 @@
+	 * @param max     a positive integer that represents the maximum
+	 *                    concatenations of the strings set
+	 * @param strings the set of strings
+	 * 
+	 * @throws IllegalArgumentException if min or max are negative numbers.
+	 */
+	public Brick(int min, int max, Set<String> strings) {
+		if (min < 0 || max < 0)
+			throw new IllegalArgumentException();
+		this.brickInterval = new IntInterval(min, max);
+		this.strings = strings;
+	}
+
+	public Brick(MathNumber min, MathNumber max, Set<String> strings) {
+		this.brickInterval = new IntInterval(min, max);
+		this.strings = strings;
+	}
+
+	/**
+	 * Builds a brick abstract element.
+	 *
+	 * @param interval an interval that yields the minimum of the brick and the
+	 *                     maximum of the brick respectively
+	 * @param strings  the set of strings
+	 */
+	public Brick(IntInterval interval, Set<String> strings) {
+		this.brickInterval = interval;
+		this.strings = strings;
+	}
+
+	@Override
+	public Brick lubAux(Brick other) throws SemanticException {
+		Set<String> resultStrings = new TreeSet<>(this.strings);
+		resultStrings.addAll(other.strings);
+
+		return new Brick(this.getMin().min(other.getMin()),
+				this.getMax().max(other.getMax()),
+				resultStrings);
+
+	}
+
+	@Override
+	public boolean lessOrEqualAux(Brick other) throws SemanticException {
+		if (this.strings.size() > other.strings.size())
+			return false;
+
+		if (other.strings.containsAll(this.strings))
+			if (this.getMin().geq(other.getMin()))
+				return this.getMax().leq(other.getMax());
+
+		return false;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
+			return false;
+		Brick brick = (Brick) o;
+		return Objects.equals(strings, brick.strings) && Objects.equals(brickInterval, brick.brickInterval);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(strings, brickInterval);
+	}
+
+	/**
+	 * Yields the min of this abstract value.
+	 * 
+	 * @return the min of this abstract value
+	 */
+	public MathNumber getMin() {
+		return this.brickInterval.getLow();
+	}
+
+	/**
+	 * Yields the max of this abstract value.
+	 * 
+	 * @return the max of this abstract value
+	 */
+	public MathNumber getMax() {
+		return this.brickInterval.getHigh();
+	}
+
+	/**
+	 * Yields the set of strings of this abstract value.
+	 * 
+	 * @return the set of strings of this abstract value
+	 */
+	public Set<String> getStrings() {
+		return strings;
+	}
+
+	@Override
+	public Brick top() {
+		return TOP;
+	}
+
+	@Override
+	public Brick bottom() {
+		return BOTTOM;
+	}
+
+	/**
+	 * Yields all the possible concatenations between min and max of the
+	 * strings. set
+	 * 
+	 * @return the set of strings with all possible concatenations between min
+	 *             and max
+	 */
+	public Set<String> getReps() {
+		if (getMin().isInfinite() || getMax().isInfinite())
+			// TODO: what to do in this case;
```

Comment:
We might want to throw a custom exception here

Reply:
We should also provide a helper method that checks if its possible to invoke this method. Something like `isFinite()` that returns true iff `!getMin().isInfinite() && !getMax().isInfinite()`. We can then use `!isFinite()` as guard of this if condition.

Reply:
I'm not sure, but I think that it wouldn't make sense to have Infinite as a minimum

Reply:
Up to you: just use the condition that makes most sense here!